### PR TITLE
Add "backgroundImageUrl" and custom css properties for controlling background of the card component

### LINKED ISSUE #2022: [Enhancement] Card Background Image
- [X] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
We should allow for background images to be used on the card component with some sort of API. 
Right now consumers have DIY'ed some solutions for having image backgrounds in card - but we can not guarantee that these will continue working, if we make changes to the card component. 

Therefore we should allow it to be applied in an easy way, where we make sure to not break that functionality.

## Describe the solution you'd like
We can either use an input property to apply it or a css custom property (or perhaps a input property that sets a custom css property!?). 

## Have you considered any alternatives?
No. 

## Are there any additional context?
#2006 is blocked by this issue. 

## Requirements & todo's: 
I've tried compiling a couple of requirements for this functionality no matter how the API is created: 
- The image should fill the entire card
- When the image is loading a blurred version of the image should be shown (might be a candidate for a spin-off issue to limit scope of this issue)
- It should be possible to change the image based on device (phone, tablet, desktop...) 

Todos: 
- Document the new API 
- Document how to change the image based on device
- Inform consumers who've hacked together this functionality that we're creating an easy way to do this. 

<hr />

### PR DESCRIPTION
**Pull request by MadsBuchmann**
## Which issue does this PR close?

This PR implements #2022

## What is the new behavior?

First two ways of setting a background image for a card has been added: 
1. The input property 'backgroundImageUrl' which just needs to be passed an url. It more or less is just a convenient way of setting the background image via javsacript. 
2. The `--kirby-card-background-image` property which allows users of the component to set the `background-image` css property of the `kirby-card` element. This allows for handling media queries & adding linear gradients etc. 

Secondly the below css custom properties has been added for controlling the background image on card: 

1. `--kirby-card-background-repeat`: Sets the 'background-repeat' property of the card
2. `--kirby-card-background-position`: Sets the 'background-position' property of the card 
3. `--kirby-card-background-size`: Sets the 'background-size' property of the card

Thirdly i've moved all of the card examples to a shared module as they were split into two separate folders before. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [X] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [X] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [X] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [X] Request that the changes are code-reviewed 
- [X] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be automatically merged to master via [automerge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request).

### CODE REVIEW

File: apps/cookbook/src/app/showcase/card-showcase/card-showcase.component.html [hunk 1]
```diff
@@ -59,14 +59,58 @@
...
   <cookbook-api-description-properties
     [properties]="properties"
   ></cookbook-api-description-properties>
-  <h4>Colors:</h4>
-  <cookbook-card-themecolor-example></cookbook-card-themecolor-example>
-  <h4>Elevation:</h4>
-  <cookbook-card-elevations-example></cookbook-card-elevations-example>
-  <h4>Clickable:</h4>
-  <cookbook-card-clickable-example></cookbook-card-clickable-example>
+  <h3>Css Custom Properties</h3>
```
Comment (jkaltoft) [inline apps/cookbook/src/app/showcase/card-showcase/card-showcase.component.html:110]:
```suggestion
  <h3>CSS Custom Properties</h3>
```

**Review by jkaltoft — COMMENTED** [review]
This looks good 👍🏻

However, I do have some questions and suggestions:

- Did you verify that it works with any valid URL for the source image?
  - [x] "Full URL" including `https://`
  - Local sources
    - [x] Absolute path, e.g. `/path/to/image.jpg`
    - [ ] Relative path, e.g. `path/to/image.jpg`
    - [x] Relative path, e.g. `../path/to/image.jpg`
  - [x] A data URI, e.g. a base64 encoded image

Either way, I suggest the tests and the examples/showcase reflect these different options.

Regarding the examples, consider showing or describing a solution/suggestion for how to use different background images on narrow (mobile) and wider (tablet) device screens.

**Comment by MadsBuchmann** [comment]
@jkaltoft - i've updated the documentation to improve the examples and explanatory text. 

I had trouble testing the 'path/to/image.jpg' url as i can only point to resources located in the assets folder. But the other relative paths were successful. Ultimately it is dependent on how the users of the component has setup their handling of static resources.

**Review by jkaltoft — APPROVED** [review]
Looks good.

Currently the base branch for the PR is `main`. Is that intentional?

**Pull request merged**